tests/mocks/nrf_rpc/mock_nrf_rpc_transport.c

```diff
@@ -0,0 +1,154 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <mock_nrf_rpc_transport.h>
+
+#include <zephyr/kernel.h>
+#include <zephyr/sys/util.h>
+#include <zephyr/ztest.h>
+
+#include <string.h>
+
+#define MAX_NUM_EXPECTED_PKTS 5
+
+typedef struct mock_nrf_rpc_tr_ctx {
+	const struct nrf_rpc_tr *transport;
+	nrf_rpc_tr_receive_handler_t receive_cb;
+	void *receive_ctx;
+
+	mock_nrf_rpc_pkt_t expected[MAX_NUM_EXPECTED_PKTS];
+	mock_nrf_rpc_pkt_t response[MAX_NUM_EXPECTED_PKTS];
```

Comment:
I wanted to tell you that it is already implemented to some extent in ZTest mocking, but it seems already deprecated :)
Let's keep your implementation then.